src/core/components/rerender/rerender.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { act, render, screen } from '@testing-library/react';
+import { type IRerenderProps, Rerender } from './rerender';
+
+describe('<Rerender /> component', () => {
+    beforeEach(() => {
+        jest.useFakeTimers();
+    });
+
+    const createTestComponent = (props?: Partial<IRerenderProps>) => {
+        return render(<Rerender {...props}>{(time) => <div>Current Time: {time}</div>}</Rerender>);
+    };
```

Comment:
Don't call the `render` function inside the `createTestComponent` function, different test might need different options to pass to the render function. Also do not rely on common properties (children prop in this case) on tests, each test should be independent. 

```suggestion
    const createTestComponent = (props?: Partial<IRerenderProps>) => {
        const completeProps: IRerenderProps = {
            children: jest.fn(),
            ...props,
        };
        
        return <Rerender {...completeProps} />;
    };
```

Reply:
✅